doc/man3/BIO_ctrl.pod

```diff
@@ -92,7 +92,7 @@ for success and -1 for failure.
 
 BIO_flush() returns 1 for success and 0 or -1 for failure.
 
-BIO_eof() returns 1 if EOF has been reached 0 otherwise.
+BIO_eof() returns 1 if EOF has been reached or an error occurred, 0 otherwise.
```

Comment:
This is really bad, 'cause now, there's no way to distinguish if EOF was reached (*reaching EOF is **not** an error!!!*) or an actual error occurred.

One way of solving this would be to add BIO_error(), which would enable the same sort of interface as feof() and ferror().

Reply:
I agree in principle, but according to @t8m the error condition does not really occur in practice (which I did not check) and at least so far users will have implicitly used `!= 0` anyway, relying on the hitherto (not adequately) documented behavior.

I suggest opening a separate issue/PR on adding `BIO_error()`.

Reply:
@levitte Please look at how the BIO_CTRL_EOF is implemented. In our BIOs a negative result can currently happen only if BIO_CTRL_EOF is unimplemented or there is no ctrl call at all in the BIO method. And all the callers IMO currently treat negative result as EOF anyway. So I do not think this is bad change at all. It just makes the current behavior more "official".